pkg/applier/applier.go

```diff
@@ -69,28 +68,47 @@ var (
 type Applier interface {
 	// Apply creates, updates, or prunes all managed resources, depending on
 	// the new desired resource objects.
-	// Returns the set of GVKs which were successfully applied and any errors.
+	// Error events are sent to the eventHandler.
+	// Returns the status of the applied objects and statistics of the sync.
 	// This is called by the reconciler when changes are detected in the
 	// source of truth (git, OCI, helm) and periodically.
-	Apply(ctx context.Context, desiredResources []client.Object) (map[schema.GroupVersionKind]struct{}, status.MultiError)
-	// Errors returns the errors encountered during apply.
-	// This method may be called while Destroy is running, to get the set of
-	// errors encountered so far.
-	Errors() status.MultiError
+	Apply(ctx context.Context, eventHandler func(Event), desiredResources []client.Object) (ObjectStatusMap, *stats.SyncStats)
```

Comment:
What about renaming `eventHandler` to `errorEventHandler` or `errorHandler` to distinguish it from `supervisorEventHandler`?

Reply:
I plan on adding inventory events to it, errors are just the first pass.

But yes, having two event handlers in the same file is a bit confusing. Open to other suggestions though. 

I was thinking maybe the supervisorEventHandler could be merged with the supervisor, but the supervisorEventHandler has a destroy flag I don't really want to pass to all the methods individually.

Reply:
> I plan on adding inventory events to it, errors are just the first pass.

I thought other inventory events would be handled by the `supervisorEventHandler`? What are other inventory events?

> but the supervisorEventHandler has a destroy flag I don't really want to pass to all the methods individually.

Only the `processWaitEvent` function reads the destroy flag. If we merge `supervisorEventHandler` with the supervisor, the flag only needs to be passed to one function, which seems not bad.

Reply:
done